### ISSUE TITLE
Derive Julia call stacks

### ISSUE THREAD

**Issue by evetion** [Deltares/Ribasim#1472]
For problems such as allocation. As discussed during a knowledge sharing session.

With a call stack it's easier to understand how a computational feature works, and how new code can be designed/integrated, or old code refactored.

edit: I envision something like https://juliadocs.org/DocumenterMermaid.jl/stable/
![Image](https://github.com/user-attachments/assets/e26da8f4-a155-4554-a4f1-66425b9006bd)

**Comment by evetion** [comment]
If this is not doable (in a sprint), let's just describe the different parts of the codebase.

**Comment by evetion** [comment]
Ok, so ideally we get a call stack in Julia that implements the AbstracTreeInterface, which we then could easily put into a markdown with mermaidjs backticks.

However, I've tested the [gist](https://gist.github.com/mkborregaard/81825c3d370bb4d8dbfe59c3b2ae4b33) partially based on the old TraceCalls and this crashed on a Ribasim.run (after ~20min). I've tested TraceFuns, and it only prints, not stores, and even that crashed (but printed a lot of nice output).

So the proper way of doing this is editing TraceCalls, or following [this documentation of Cassette.jl](https://julia.mit.edu/Cassette.jl/stable/contextualdispatch/) to actually keep track of all the calls.

However, that seems out of scope here, so I would propose to do a manual documentation of the call stack for:
- toml/DB loads
- all the different callbacks
- water_balance! specific calls
